TH-1437 | Clean up test output

karisal-anders opened the pull request:
## Description

Clean up events-helsinki app's tests' output.

Only remaining errors/warnings in events-helsinki app's tests' output are:
1. `Warning: Each child in a list should have a unique "key" prop.`
   - Check the render method of `CarouselSlider`. See https://reactjs.org/link/warning-keys for more information.
2. `Warning: validateDOMNesting(...): <a> cannot appear as a descendant of <a>.`

Fixing these errors/warning may need fixes to [RHHC](https://github.com/City-of-Helsinki/react-helsinki-headless-cms/)

This PR is **quite similar** to **PR #870** and **PR #871**.

## Related

[TH-1437](https://helsinkisolutionoffice.atlassian.net/browse/TH-1437)

[TH-1437]: https://helsinkisolutionoffice.atlassian.net/browse/TH-1437?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

karisal-anders committed:
feat(tests): enable readable apollo client errors in events-helsinki

Previously tests showed errors like:
```
An error occurred! For more details, see the full error text at
https://go.apollo.dev/c/err#%7B...
```
where the link went on for many rows.

This enables the apollo client errors messages to be directly readable
rather than having to click on a link and go to a webpage where one can
see the error message.

refs TH-1437

karisal-anders committed:
fix(tests): hide sourcemap missing warnings in events-helsinki tests

This declutters output of apps/events-helsinki tests.

refs TH-1437

karisal-anders committed:
fix(tests): cleanup events AppConfig tests & their output

Previously running events AppConfig.test.tsx tests there was output:
```
stdout | src/domain/app/__tests__/AppConfig.test.tsx > provides number
config 'defaultRevalidate'
Could not parse env value SyntaxError: Unexpected token 'S', "Some
string" is not valid JSON
```

This removes that output from running the tests & instead tests for it.

Also remove all @ts-ignore from AppConfig.test.tsx & fix the related
typings.

refs TH-1437

karisal-anders committed:
fix(tests): hide warning caused by HDS 3's Select component

Previously when running TargetAgeGroupSelector tests there was ~warning:
```
stderr | src/domain/search/eventSearch/advancedSearch/__tests__/
TargetAgeGroupSelector.test.tsx > <TargetAgeGroupSelector /> > renders
without crashing

Warning: Li: Support for defaultProps will be removed from function
components in a future major release. Use JavaScript default parameters
instead.
    at Li (node_modules\hds-react\cjs\index.js:1:318591)
    at TargetAgeGroupSelector (apps\events-helsinki\src\domain\search\
    eventSearch\advancedSearch\TargetAgeG...
```

Hide this warning as there's nothing we can do to not have this warning
except upgrade to HDS v4, which is planned to be done during late 2025,
and hope that HDS v4 fixes this, or hide it. Opting for the latter here.

refs TH-1437

karisal-anders committed:
fix(tests): use unique IDs in events EventPageContainer tests

This change removes Apollo client cache warning when running events
EventPageContainer tests.

refs TH-1437

karisal-anders committed:
fix(tests): simplify events EventPageContainer's keywordless event test

refs TH-1437

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr872.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-sports-pr872.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-hobbies-pr872.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr872.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**LIIKUNTA-UI branch is deployed to platta: https://liikunta-pr872.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr872.dev.hel.ninja :laughing::tada::tada::tada:

karisal-anders committed:
fix(tests): fix events Search "for accessibility violations" test

Also remove 50s timeout from test case as unnecessary.

Previously running events Search.test.tsx tests there was ~output:
```
$ yarn test src/domain/search/eventSearch/advancedSearch/__tests__/
Search.test.tsx --watch

stderr | src/domain/search/eventSearch/advancedSearch/__tests__/
Search.test.tsx > for accessibility violations

Error: Not implemented: HTMLCanvasElement.prototype.getContext (without
installing the canvas npm package)
...

Warning: An update to PlaceSelector inside a test was not wrapped in
act(...).

When testing, code that causes React state updates should be wrapped
into act(...):

act(() => {
  /* fire events that update state */
});
/* assert on the output */

This ensures that you're testing the behavior the user would see in the
browser. Learn more at https://reactjs.org/link/wrap-tests-with-act
	at PlaceSelector (packages\components\src\components\domain\place\
	placeSelector\PlaceSelector.tsx...
```

refs TH-1437

karisal-anders commented:
[Force pushed](https://github.com/City-of-Helsinki/events-helsinki-monorepo/compare/60f6c1572a2b464eb11f1844283bf3292d15d260..7762d8d70ec5d4826676b8df97903b44d5b4559f):
- Reordered commits, simplified the now last commit by removing unnecessary canvas package install and instead adding mocking into the vitest-axe accessibility violations testing test case.

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-sports-pr872.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr872.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**LIIKUNTA-UI branch is deployed to platta: https://liikunta-pr872.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr872.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr872.dev.hel.ninja :laughing::tada::tada::tada:
